src/main/java/com/uber/cadence/internal/replay/ClockDecisionContext.java

```diff
@@ -271,4 +191,37 @@ void handleMarkerRecorded(HistoryEvent event) {
       log.warn("Unexpected marker: " + event);
     }
   }
+
+  int getVersion(
+      String changeId,
+      Func1<MutableSideEffectData, byte[]> markerDataSerializer,
+      Func1<byte[], MutableSideEffectData> markerDataDeserializer,
+      int minSupported,
+      int maxSupported) {
+    Optional<byte[]> result =
+        versionHandler.handle(
+            changeId,
+            markerDataSerializer,
+            markerDataDeserializer,
+            (stored) -> {
+              if (stored.isPresent()) {
+                return Optional.empty();
+              }
+              return Optional.of(ByteBuffer.allocate(4).putInt(maxSupported).array());
```

Comment:
Why are we using binary serialization here? Let's use DataConverter through a lambda passed here.